client/hijack.go

```diff
@@ -72,7 +74,21 @@ func (cli *Client) postHijacked(ctx context.Context, path string, query url.Valu
 	defer clientconn.Close()
 
 	// Server hijacks the connection, error 'connection closed' expected
-	_, err = clientconn.Do(req)
+	resp, err := clientconn.Do(req)
+	if err != nil {
+		return types.HijackedResponse{}, err
+	}
+	if resp != nil {
+		if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusSwitchingProtocols {
+			defer resp.Body.Close()
+
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return types.HijackedResponse{}, err
+			}
+			return types.HijackedResponse{}, fmt.Errorf("Error response from daemon: %s", bytes.TrimSpace(body))
+		}
+	}
```

Comment:
What do you think about something like this:

```go

defer ensureReaderClosed(resp)

switch resp.StatusCode {
case http.StatusOK, http.StatusSwitchingProtocols:
    rwc, br := clientconn.Hijack()
    return types.HijackedResponse{Conn: rwc, Reader: br}, nil
}

body, err := ioutil.ReadAll(resp.Body)
...
return types.HijackedResponse{}, fmt.Errorf("...")
```

I think you want the response closed called in both cases?